backend/pkg/commons/db2/raw/client_test.go

```diff
@@ -0,0 +1,307 @@
+				t.Errorf("got %v, want %v", got, want)
+			}
+
+			receipts, err := ethClient.BlockReceipts(context.Background(), rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(tt.block)))
+			if err != nil {
+				t.Fatalf("BlockReceipts() error = %v", err)
+			}
+			if len(block.Transactions()) != 0 && len(receipts) == 0 {
+				t.Errorf("receipts should not be empty")
+			}
+
+			var traces []GethTraceCallResultWrapper
+			if err := rpcClient.Call(&traces, "debug_traceBlockByNumber", hexutil.EncodeBig(block.Number()), gethTracerArg); err != nil {
+				t.Fatalf("debug_traceBlockByNumber() error = %v", err)
+			}
+			if len(block.Transactions()) != 0 && len(traces) == 0 {
+				t.Errorf("traces should not be empty")
+			}
+		})
+	}
+}
+
+func benchmarkBlockRetrieval(b *testing.B, ethClient *ethclient.Client, rpcClient *rpc.Client) {
+	b.ResetTimer()
+	for j := 0; j < b.N; j++ {
+		blockTestNumber := int64(20978000 + b.N)
+		_, err := ethClient.BlockByNumber(context.Background(), big.NewInt(blockTestNumber))
+		if err != nil {
+			b.Fatalf("BlockByNumber() error = %v", err)
+		}
+
+		if _, err := ethClient.BlockReceipts(context.Background(), rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(blockTestNumber))); err != nil {
+			b.Fatalf("BlockReceipts() error = %v", err)
+		}
+
+		var traces []GethTraceCallResultWrapper
+		if err := rpcClient.Call(&traces, "debug_traceBlockByNumber", hexutil.EncodeBig(big.NewInt(blockTestNumber)), gethTracerArg); err != nil {
+			b.Fatalf("debug_traceBlockByNumber() error = %v", err)
+		}
+	}
+}
+
+func BenchmarkErigonNode(b *testing.B) {
+	node := os.Getenv("ETH1_ERIGON_ENDPOINT")
+	if node == "" {
+		b.Skip("skipping test, please set ETH1_ERIGON_ENDPOINT")
+	}
+
+	rpcClient, err := rpc.DialOptions(context.Background(), node)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	benchmarkBlockRetrieval(b, ethclient.NewClient(rpcClient), rpcClient)
+}
+
+func BenchmarkRawBigTable(b *testing.B) {
+	project := os.Getenv("BIGTABLE_PROJECT")
+	instance := os.Getenv("BIGTABLE_INSTANCE")
+	if project == "" || instance == "" {
+		b.Skip("skipping test, set BIGTABLE_PROJECT and BIGTABLE_INSTANCE")
+	}
+
+	bt, err := database.NewBigTable(project, instance, nil)
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	rawStore := WithCache(NewStore(database.Wrap(bt, BlocksRawTable, "")))
+	rpcClient, err := rpc.DialOptions(context.Background(), "http://foo.bar", rpc.WithHTTPClient(&http.Client{
+		Transport: NewBigTableEthRaw(rawStore, chainID),
+	}))
+	if err != nil {
+		b.Fatal(err)
+	}
+
+	benchmarkBlockRetrieval(b, ethclient.NewClient(rpcClient), rpcClient)
+}
+
+func BenchmarkAll(b *testing.B) {
+	b.Run("BenchmarkErigonNode", func(b *testing.B) {
+		BenchmarkErigonNode(b)
+	})
+	b.Run("BenchmarkRawBigTable", func(b *testing.B) {
+		BenchmarkRawBigTable(b)
+	})
+}
+
+func TestBigTableClient(t *testing.T) {
+	tests := []struct {
+		name  string
+		block FullBlockData
+	}{
+		{
+			name:  "test block",
```

Comment:
Similar here, we can consider renaming it to "Test single block" and below to "Test single block with two uncles"